flair/hyperparameter/param_selection.py

```diff
@@ -0,0 +1,260 @@
+    ):
+        """
+        :param corpus: the corpus
+        :param tag_type: tag type to use
+        :param base_path: the path to the result folder (results will be written to that folder)
+        :param max_epochs: number of epochs to perform on every evaluation run
+        :param evaluation_metric: evaluation metric used during training
+        :param training_runs: number of training runs per evaluation run
+        :param optimization_value: value to optimize
+        """
+        super().__init__(
+            corpus,
+            base_path,
+            max_epochs,
+            evaluation_metric,
+            training_runs,
+            optimization_value,
+        )
+
+        self.tag_type = tag_type
+        self.tag_dictionary = self.corpus.make_label_dictionary(self.tag_type)
+
+    def _set_up_model(self, params: dict):
+        sequence_tagger_params = {key: params[key] for key in params if key in SEQUENCE_TAGGER_PARAMETERS}
+
+        tagger: SequenceTagger = SequenceTagger(
+            tag_dictionary=self.tag_dictionary,
+            tag_type=self.tag_type,
+            **sequence_tagger_params,
+        )
+        return tagger
+
+
+class TextClassifierParamSelector(ParamSelector):
+    def __init__(
+        self,
+        corpus: Corpus,
+        label_type: str,
+        multi_label: bool,
+        base_path: Union[str, Path],
+        document_embedding_type: str,
+        max_epochs: int = 50,
+        evaluation_metric: EvaluationMetric = EvaluationMetric.MICRO_F1_SCORE,
+        training_runs: int = 1,
+        optimization_value: OptimizationValue = OptimizationValue.DEV_LOSS,
+    ):
+        """
+        :param corpus: the corpus
+        :param label_type: string to identify the label type ('question_class', 'sentiment', etc.)
+        :param multi_label: true, if the dataset is multi label, false otherwise
+        :param base_path: the path to the result folder (results will be written to that folder)
+        :param document_embedding_type: either 'lstm', 'mean', 'min', or 'max'
+        :param max_epochs: number of epochs to perform on every evaluation run
+        :param evaluation_metric: evaluation metric used during training
+        :param training_runs: number of training runs per evaluation run
+        :param optimization_value: value to optimize
+        """
+        super().__init__(
+            corpus,
+            base_path,
+            max_epochs,
+            evaluation_metric,
+            training_runs,
+            optimization_value,
+        )
+
+        self.multi_label = multi_label
+        self.label_type = label_type
+        self.document_embedding_type = document_embedding_type
+
+        self.label_dictionary = self.corpus.make_label_dictionary(self.label_type)
+
+    def _set_up_model(self, params: dict):
+        embdding_params = {key: params[key] for key in params if key in DOCUMENT_EMBEDDING_PARAMETERS}
+
+        if self.document_embedding_type == "lstm":
```

Comment:
it seems the only options are RNN or pooled word embeddings, but the standard approach today would be to simply use a transformer only which far outperforms RNNs trained over word embeddings. Not necessarily something that needs to be fixed in this PR but something that limits the usefulness of our parameter selection for text classifiers at the moment.

Reply:
As per your suggestion (which makes perfects sense - it felt really weird readding this outdated RNN code) I refactored `TextClassifierParamSelector` so that it uses `TransformerDocumentEmbeddings` now. I documented it to a reasonable extent in the tutorial.